UniWatch/DataAccess/DataAccess.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using UniWatch.Models;
+
+namespace UniWatch.DataAccess
+{
+    public class DataAccess : IDataAccess
+    {
+        private AppDbContext _db;
+        private IClassManager _classManager;
+        private ILectureManager _lectureManager;
+        private IStudentManager _studentManager;
+
+        /// <summary>
+        /// The class manager for class related functionality
+        /// </summary>
+        public IClassManager ClassManager
+        {
+            get
+            {
+                if(_classManager == null)
```

Comment:
You probably know this, but you can perform a null check in-line. Nothing is wrong here.

`return _classManager ?? (_classManager = new ClassManager(_db));`